gt/communication/ghex_comm.cpp

```diff
@@ -0,0 +1,37 @@
+#include "./ghex_comm.hpp"
+
+#include <ghex/glue/gridtools/field.hpp>
+
+namespace communication {
+
+namespace ghex_comm {
+
+std::function<void(storage_t &)>
+comm_halo_exchanger(grid const &g, storage_t::storage_info_t const &sinfo) {
+  static std::unique_ptr<grid::comm_obj_type> comm_obj;
```

Comment:
I wouldn’t bother about reuse here, but go for simplicity. In the final benchmark application, most probably this function will be called only once anyway.